content_scripts/mode_find.js

```diff
@@ -194,34 +192,42 @@ class FindMode extends Mode {
       return;
     }
 
-    // innerText will not return the text of hidden elements, and strip out tags while preserving
-    // newlines.
-    // NOTE(mrmr1993): innerText doesn't include the text contents of <input>s and <textarea>s. See
-    // #1118.
-    const text = document.body.innerText;
-    const regexMatches = text.match(pattern);
-    if (this.query.isRegex) {
-      this.query.regexMatches = regexMatches;
-    }
-
-    if (this.query.isRegex) {
-      this.query.activeRegexIndex = 0;
-    }
+    const textNodes = getAllTextNodes();
+    const matchedNodes = textNodes.filter((node) => {
+      return node.textContent.match(pattern);
+    });
+    const regexMatches = matchedNodes.map((node) => node.textContent.match(pattern));
+    this.query.regexMatches = regexMatches;
+    this.query.regexPattern = pattern;
+    this.query.regexMatchedNodes = matchedNodes;
+    this.query.activeRegexIndices = [0, 0];
 
-    return this.query.matchCount = regexMatches != null ? regexMatches.length : null;
+    return this.query.matchCount = regexMatches != null ? regexMatches.flat().length : null;
   }
 
   static getNextQueryFromRegexMatches(stepSize) {
```

Comment:
I had to look at the caller to understand what stepSize was supposed to be, and why it wouldn't always be "1". Instead let's use the parameter `backwards` to make this function match the rest of the code, and calculate stepSize inside of `getNextQueryFromRegexMatches` based on the value of `backwards`.